system/Validation/Rules.php

```diff
@@ -352,13 +353,18 @@ public function required($str = null): bool
 	 *     required_with[password]
 	 *
 	 * @param string|null $str
-	 * @param string      $fields List of fields that we should check if present
+	 * @param string|null $fields List of fields that we should check if present
 	 * @param array       $data   Complete list of fields from the form
 	 *
 	 * @return boolean
 	 */
-	public function required_with($str = null, string $fields, array $data): bool
+	public function required_with($str = null, string $fields = null, array $data = []): bool
 	{
+		if (is_null($fields) || empty($data))
+		{
+			throw new InvalidArgumentException('You must supply the parameters: fields, data.');
+		}
```

Comment:
```suggestion
	public function required_with($str = null, string $fields = '', array $data = []): bool
	{
		if ($fields === '' || $data === [])
		{
			throw new InvalidArgumentException('You must supply the parameters: fields, data.'); // @codeCoverageIgnore
		}
```

Reply:
Same as below, `$fields` is allowed to be an empty string so we need to add the `null` option. This could probably be addressed on calling classes but it is also possible that someone is calling this directly (though I really hope not).